### PR TITLE
Correct optional datastructure file RDG move paths

### PR DESCRIPTION
**Pull request by e-mcginnis**
just another reason getting the list of rdg files needs a test! But thats going to have to wait for now as it will be non-trivial to implement...

added some useful loglines as well

### CODE REVIEW

**Comment by yshalabi** [comment]
> just another reason getting the list of rdg files needs a test! But thats going to have to wait for now as it will be non-trivial to implement...
> 
> added some useful loglines as well

I tried to add a test for this in katana-enterprise #2093 but when the underlying interface was refactored before I could merge I decided to close it and just revisit later when the interface stabilizes.... of course I forgot. IDK if there is anything useful there but you can check it out to see if it is worth resuscitating.

**Comment by e-mcginnis** [comment]
I'm thinking it will look a bit like the storage format version test, basically check the actual on disk version of the maximal rdg against the FileNames output

File: libtsuba/src/RDGManifest.cpp [hunk 1]
```diff
@@ -286,7 +287,7 @@ katana::RDGManifest::FileNames() {
       }
 
       for (auto it : header.optional_datastructure_manifests()) {
-        fnames.emplace(it.first);
+        fnames.emplace(it.second);
```
Comment (yshalabi) [inline libtsuba/src/RDGManifest.cpp:290]: These kinds of bugs are harder to make if we have wrappers around the iterator type which return the desired value. e.g: FileNameFrom...

File: libtsuba/src/RDGPartHeader.cpp [hunk 2]
```diff
@@ -345,10 +345,14 @@ katana::to_json(json& j, const katana::RDGPartHeader& header) {
         "RDGPartHeader.unstable_storage_format_ is true");
   }
 
-  KATANA_LOG_DEBUG(
+  KATANA_LOG_WARN(
```
Comment (yshalabi) [inline libtsuba/src/RDGPartHeader.cpp:348]: helpful stuff I think

**Review by yshalabi — APPROVED** [review]
LGTM

File: libtsuba/src/RDGManifest.cpp (hunk 1 above)
Comment (pyrito) [reply libtsuba/src/RDGManifest.cpp:290]: Dang

**Pull request merged**